### ISSUE TITLE
Object store builder blocks on IO when building

### ISSUE THREAD

**Issue by chebbyChefNEQ** [apache/arrow-rs#6469]
**Describe the bug**
Cloud based (tested for GCS) store builder blocks on file IO synchronously in `.build`. This happens by:
`object_store` -> `reqwest::ClientBuild::build` -> `rustls::load_native_cert`

**To Reproduce**
```rust
use object_store::gcp::GoogleCloudStorageBuilder;

fn main() {
    for _ in 0..100 {
        let start = std::time::Instant::now();
        let _gcs = GoogleCloudStorageBuilder::from_env().with_bucket_name("some-bucket").build().unwrap();
        println!("built in: {:?}", start.elapsed());
    }
}
```

**Expected behavior**
calling `build` should return a future and not block

**Additional context**
I'll create a ticket in `reqwest` as well, but maybe we could capture the client build in `tokio::spawn_blocking` for the time being?

**Comment by chebbyChefNEQ** [comment]
![linux](https://github.com/user-attachments/assets/df3d47b1-4932-4fd3-bade-a8c0eb158930)
See flame graph showing `std::fs` in the stack

**Comment by chebbyChefNEQ** [comment]
upon further testing this is cause by the feature `rustls-tls-native-roots`. `default-tls` does not have this behavior

**Comment by chebbyChefNEQ** [comment]
issue created in reqwest https://github.com/seanmonstar/reqwest/issues/2437

**Comment by tustvold** [comment]
FWIW the various object store implementations are designed to be created once and then reused, otherwise things like connection pooling, etc... won't work correctly and you will pay for non-trivial setup costs such as loading certificates

**Comment by tustvold** [comment]
> I def lack the context on what we need native roots maybe that's something we could change

There is some further context on this specific issue - https://github.com/apache/arrow-rs/issues/4870

TLDR you can opt-in to using webpki roots, or native-tls depending on your preferences.

> our code makes a bit hard to do DI style constructions

I'd strongly encourage you to try, polars started reporting very peculiar issues a while back (https://github.com/pola-rs/polars/issues/14384) that was ended up being related to it spinning up hundreds of separate clients. In the end they used a global cache for this and the issues went away - https://github.com/pola-rs/polars/pull/14598

**Comment by tustvold** [comment]
I'm going to close this, as reading the linked upstream ticket I am not sure there is anything planned in this repository
